src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs

```diff
@@ -351,6 +351,18 @@ void UpdateContent()
 			if (content == null || _mauiContext == null)
 				return;
 
+			if (content is Border border)
+			{
+				foreach (var child in border.LogicalChildrenInternal)
+				{
+					if (child is IShape)
+					{
+						((VisualElement)child).IsVisible = false;
```

Comment:
Could you share more details about this change?

Reply:
@jsuarezruiz When we set the property `IsVisible` to `false` then the parent layout doesn't consider this shape when calculating the required size for its content.

https://github.com/kubaflo/maui/blob/9d50c370103718c654d67d51fc13827719f125fa/src/Controls/src/Core/Layout.cs#L663

I couldn't find a better fix for that. From what I understand `Frame` is rather a legacy component, so sometimes it has an unexpected behavior. I don't use it in my code, but many developers do and they often are not aware that it is better to use border to fix many issues. Therefore, I believe that this bug should be fixed to save them from being annoyed 😅

Reply:
Can we just set `DisableLayout` to true?
https://github.com/dotnet/maui/blob/848202887c7a35d1250c4cb518e899ec65347545/src/Controls/src/Core/Layout.cs#L623

Reply:
@PureWeen I tried but it doesn't work:
```
                       if (content is Border border)
			{
				var shape = border.LogicalChildrenInternal.FirstOrDefault(child => child is IShape);
				if (shape is VisualElement visualElement)
				{
					visualElement.DisableLayout = true; // doesn't work
					visualElement.IsVisible = false; // works
				}
			}
```